src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -2345,7 +2395,128 @@ public static void copyImplementation(Design src, Design dest, boolean lockPlace
 					p.setIsPIPFixed(true);
 				}
 			}
-		}		
+		}
+
+		if (copyStaticNets) {
+			copyStaticNets(dest, staticNets, siteInstsOfCells);
+		}
+	}
+
+	/**
+	 * Copy the route of static nets feeding the sinks within the given SiteInst.
+	 * @param dest The destination design
+	 * @param staticNets The list of static nets to copy
+	 * @param siteInstsOfCells The set of SiteInst containing the sinks of the static nets
+	 */
+	private static void copyStaticNets(Design dest, List<Net> staticNets, Set<SiteInst> siteInstsOfCells) {
+
+		// build map to from site_pin to list of bels
+		HashMap<String, List<String>> sitePin2Bels = new HashMap<String,List<String>>()
+		{{
+			put("A_O",  Arrays.asList("A5LUT", "A6LUT"));
+			put("AMUX", Arrays.asList("A5LUT", "A6LUT"));
+			put("B_O",  Arrays.asList("B5LUT", "B6LUT"));
+			put("BMUX", Arrays.asList("B5LUT", "B6LUT"));
+			put("C_O",  Arrays.asList("C5LUT", "C6LUT"));
+			put("CMUX", Arrays.asList("C5LUT", "C6LUT"));
+			put("D_O",  Arrays.asList("D5LUT", "D6LUT"));
+			put("DMUX", Arrays.asList("D5LUT", "D6LUT"));
+			put("E_O",  Arrays.asList("E5LUT", "E6LUT"));
+			put("EMUX", Arrays.asList("E5LUT", "E6LUT"));
+			put("F_O",  Arrays.asList("F5LUT", "F6LUT"));
+			put("FMUX", Arrays.asList("F5LUT", "F6LUT"));
+			put("G_O",  Arrays.asList("G5LUT", "G6LUT"));
+			put("GMUX", Arrays.asList("G5LUT", "G6LUT"));
+			put("H_O",  Arrays.asList("H5LUT", "H6LUT"));
+			put("HMUX", Arrays.asList("H5LUT", "H6LUT"));
+		}};
+
+		// Map from a node to its driver PIP
```

Comment:
This functionality looks a lot like `DesignTools.getTrimmablePIPsFromPins()`, which takes a `Net` and a set of `SitePinInst`-s, and returns a set of PIPs necessary to feed those. Is it possible to reuse that code?

https://github.com/Xilinx/RapidWright/blob/7aafb67e3460f5373d0911c722cc39764d039efc/src/com/xilinx/rapidwright/design/DesignTools.java#L1107

Reply:
There is at least one difference.  Let's consider a route from one GND source (S) to two sinks (T1 and T2) and the last common node/pip is X.  If only T1 is in a cell to be copied, the tracing inside copyStaticNets will extract all pips from S to T1. However, based on the description of getTrimmablePIPsFromPins, only the pips from X to T1 are returned.

Reply:
Ah yes, you're right! Let's jot that down as a comment so I don't forget about it in the future.